src/bootstrap/configure.py

```diff
@@ -495,132 +509,170 @@ def apply_args(known_args, option_checking, config):
+        targets.add(config["build"]["build"])
+    else:
+        targets.add(bootstrap.default_build_triple(verbose=False))
 
-    # Fill out the `targets` array by giving all configured targets a copy of the
-    # `target` section we just loaded from the example config
-    configured_targets = [build(known_args)]
     if "build" in config:
         if "host" in config["build"]:
-            configured_targets += config["build"]["host"]
+            targets.update(config["build"]["host"])
         if "target" in config["build"]:
-            configured_targets += config["build"]["target"]
+            targets.update(config["build"]["target"])
     if "target" in config:
         for target in config["target"]:
-            configured_targets.append(target)
-    for target in configured_targets:
-        targets[target] = sections["target"][:]
-        # For `.` to be valid TOML, it needs to be quoted. But `bootstrap.py` doesn't use a proper TOML parser and fails to parse the target.
-        # Avoid using quotes unless it's necessary.
-        targets[target][0] = targets[target][0].replace(
-            "x86_64-unknown-linux-gnu",
-            "'{}'".format(target) if "." in target else target,
-        )
+            targets.add(target)
+    return list(targets)
 
-    if "profile" not in config:
-        set("profile", "dist", config)
-    configure_file(sections, top_level_keys, targets, config)
-    return section_order, sections, targets
+
+def write_block(f, config, block):
+    last_line = block[-1]
+    key = last_line.split("=")[0].strip(" #")
+    value = config[key] if key in config else None
+    if value is not None:
+        for ln in block[:-1]:
+            f.write(ln + "\n")
+        f.write("{} = {}\n".format(key, to_toml(value)))
+
+
+def write_section(f, config, section_lines):
+    block = []
+    for line in section_lines[1:]:
+        if line.count("=") == 1:
+            block.append(line)
+            write_block(f, config, block)
+            block = []
+        else:
+            block.append(line)
+
+
+# Write out the configuration toml file to f, using config.example.toml as a
+# template.
+def write_config_toml(f, config):
+    with open(rust_dir + "/config.example.toml") as example_config:
+        lines = example_config.read().split("\n")
+
+    section_name = None
+    section = []
+    block = []
+
+    i = 0
+    # Drop the initial comment block
+    for line in lines:
+        if not line.startswith("#"):
+            break
+        i += 1
+
+    for line in lines[i:]:
+        if line.startswith("["):
+            if section:
+                # Write out the previous section before starting a new one.
+                #
+                # Note that the `target` section is handled separately as we'll
+                # duplicate it per configured target, so there's a bit of special
+                # handling for that here.
+                if section_name.startswith("target"):
```

Comment:
This only writes out a section when we encounter the next, which means the final section is never written (i.e. `dist` is dropped)

Reply:
Oops, you're totally right. Will fix.